src/main/java/com/timgroup/statsd/Message.java

```diff
@@ -119,9 +116,10 @@ public boolean getDone() {
     @Override
     public int hashCode() {
         // cache it
-        if (this.hash == null) {
-            this.hash = new Integer(Objects.hash(this.aspect)) * HASH_MULTIPLIER;
-            this.hash += Objects.hash(this.tags);
+        if (this.hash == Integer.MIN_VALUE) {
+            this.hash = type.hashCode() * HASH_MULTIPLIER * HASH_MULTIPLIER
+                    + aspect.hashCode() * HASH_MULTIPLIER
+                    + Arrays.hashCode(this.tags);
```

Comment:
Would this be equivalent?

```suggestion
            this.hash = Objects.hash(type, aspect, tags);
```

Reply:
This allocates a varargs array which does not get scalarized except on very recent versions of OpenJDK.

Reply:
Indeed. Considering that this will be a particularly hot code it is good not to prevent JIT from optimizing it as much as possible.

Reply:
Since this is running inside the customer application, we need to be careful about any overhead.  Increasingly customers are concerned about p99 latency (e.g. GC overhead), so it is good to get rid of any unnecessary allocation.

Additionally, this allocation is actually more expensive best case than the if check mentioned above.